apt/minimization/minimizer.py

```diff
@@ -893,7 +857,11 @@ def _generalize_from_tree(self, original_data, prepared_data, level_nodes, cells
 
     def _generalize_indexes(self, original_data, cells, all_indexes):
         # prepared data include one hot encoded categorical data + QI
-        representatives = pd.DataFrame(columns=self._features)  # empty except for columns
+        dtypes = original_data.dtypes.to_dict()
+        new_dtypes = {}
+        for t in dtypes.keys():
+            new_dtypes[t] = pd.Series(dtype=dtypes[t].name)
+        representatives = pd.DataFrame(new_dtypes)  # empty except for columns
```

Comment:
I'm not sure it gives you the correct columns. There will be only one column per data type.